src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,202 @@
+	// geocoding specid=fic
+	private static final String GEOCODING_LIMIT_PARAMETER = "limitnbresult";
+	// search
+	private static final String SEARCH_LIMIT_PARAMETER = "to";
+	private static final String SEARCH_QUERY_PARAMETER = "q";
+
+
+	private final String geocodingUrl;
+	private final String reverseGeocodingUrl;
+	private final String searchURL;
+	private final String apiKey;
+	private final Client jerseyClient;
+
+	public ConverterResourceGisgraphy(String geocodingUrl,
+			String reverseGeocodingUrl, String searchURL, String apiKey,
+			Client jerseyClient) {
+		this.geocodingUrl = geocodingUrl;
+		this.reverseGeocodingUrl = reverseGeocodingUrl;
+		this.searchURL = searchURL;
+		this.jerseyClient = jerseyClient;
+		this.apiKey = apiKey;
+	}
+
+	@GET
+	@Timed
+	public Response handle(@QueryParam("q") @DefaultValue("") String query,
+			@QueryParam("point") @DefaultValue("") String point,
+			@QueryParam("radius") @DefaultValue("") String radius,
+			@QueryParam("country") @DefaultValue("") String country,
+			@QueryParam("limit") @DefaultValue("5") Integer limit,
+			@QueryParam("reverse") @DefaultValue("false") boolean reverse,
+			@QueryParam("autocomplete") @DefaultValue("false") boolean autocomplete) {
+		limit = fixLimit(limit);
+		String lat = null;
+		String lng = null;
+		if (!point.isEmpty()){
+			String[] cords = point.split(",");
+			 lat = cords[0];
+			 lng = cords[1];
+		}
+		checkParameters(query, reverse, autocomplete, lat, lng);
+
+		WebTarget target;
+		if (reverse) {
+			target = buildReverseGeocodingTarget(query, lat, lng, radius,
+					country, limit);
+		} else if (autocomplete) {
+			target = buildAutocompleteTarget(query, lat, lng, radius, country,
+					limit);
+		} else {
+			target = buildGeocodingTarget(query, lat, lng, radius, country,
+					limit);
+		}
+
+		Response response = target.request().accept("application/json").get();
+		Status status = new Status(response.getStatus(), response
+				.getStatusInfo().getReasonPhrase());
+		failIfResponseNotSuccessful(target, status);
+
+
+		try {
+			if (!autocomplete) {
+				GisgraphyGeocodingResult feed = response
+						.readEntity(GisgraphyGeocodingResult.class);
+				return Converter.convertFromGisgraphyList(feed.result, status);
+			}
+			else {
+				GisgraphySearchResult feed = response
+						.readEntity(GisgraphySearchResult.class);
+				return Converter.convertFromGisgraphySearchList(feed.getResponse(), status);
+			}
+		} catch (Exception e) {
+			LOGGER.error("There was an issue with the target "
+					+ target.getUri() + " the provider returned: "
+					+ status.code + " - " + status.message);
+			throw new BadRequestException(
+					"error deserializing geocoding feed");
+		}
+	}
+
+	protected void processFeedException(WebTarget target, Status status) {
+		
+	}
+
+	private void checkParameters(String query, boolean reverse, boolean autocomplete, String lat,String lng) {
+		if (query!=null && query.trim().isEmpty() && !reverse){
```

Comment:
Nice, the trim is a nice idea, haven't used that in `AbstractConverterResource#checkInvalidParameter`. Do you think it would make sense to reuse `AbstractConverterResource#checkInvalidParameter` and just add the autocomplete part to the check? I would like to avoid code duplication.

```
private checkInvalidParameter(boolean reverse, String query, String point, boolean autocomplete) {
   super.checkInvalidParameter(reverse, query, point);
  if (reverse && autocomplete){
    throw new BadRequestException("autocomplete is not available in reverse geocoding request, set reverse or autocomplete to false but not both");
}   
``` 

We could then move this code after the check, as this could create a index out of bounds exception for non empty points that don't contain a ' ,'?:
```
		String lat = null;
		String lng = null;
		if (!point.isEmpty()){
			String[] cords = point.split(",");
			 lat = cords[0];
			 lng = cords[1];
		}
```

Reply:
ok but the AbstractConverterResource#checkInvalidParameter has not the same parameters as ConverterResourceGisgraphy##checkInvalidParameter. Do you want me to modify the signature of AbstractConverterResource ?I don't think add the boolean autocomplete make sense since it means nothing for the AbstractConverterResource. but tell me (we are not force to use the autocomplete parameter)